docs/storage/gcsupload/gcsupload_test.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2017 Google Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"strings"
+	"testing"
+
+	"golang.org/x/net/context"
+
+	"cloud.google.com/go/storage"
+
+	"github.com/GoogleCloudPlatform/golang-samples/internal/testutil"
+)
+
+func TestUpload(t *testing.T) {
+	tc := testutil.SystemTest(t)
+
+	client, bh, cleanupBucket := withBucketCleanup(t, tc.ProjectID, "gcs-upload-test")
```

Comment:
I can't clean up at the beginning of the test because I need the resources that are returned.
And also the concern of t.Fatal* not cleaning up doesn't seem to be a problem, t.Fatal* seem to run defers alright, please see https://play.golang.org/p/tMOdt0MTTf.